Add e2e RBD and iSCSI volume tests.

jsafrane opened the pull request:
The test containers needs to run with --privileged, therefore it's disabled by default.

Tested on Fedora 21 and Ubuntu 14.04 as hosts.

brendandburns commented on contrib/for-tests/volumes-tester/iscsi/Makefile:
```diff
@@ -0,0 +1,24 @@
+	# Push version tagged image to repository (since this image is already pushed it will simply create or update version tag)
+	gcloud docker push gcr.io/google_containers/volume-iscsi:$(TAG)
+
+clean:
```
need something here?

jsafrane replied:
Not really, I'm removing it.

brendandburns commented:
LGTM.  I know nothing of Ceph and even less of iSCSI, but good enough to merge for testing purposes.

Do you need someone on the Google side to push the gcr.io/google_containers/... images?

jsafrane commented:
> Do you need someone on the Google side to push the gcr.io/google_containers/... images?

Yes please, @saad-ali did it for my previous NFS and Gluster images.

mikedanese commented:
@k8s-bot ok to test.

mikedanese commented:
This is not compiling now for some reason.

brendandburns commented:
```sh
# github.com/GoogleCloudPlatform/kubernetes/test/e2e
_output/local/go/src/github.com/GoogleCloudPlatform/kubernetes/test/e2e/volumes.go:369: pod.Status undefined (type string has no field or method Status)
_output/local/go/src/github.com/GoogleCloudPlatform/kubernetes/test/e2e/volumes.go:416: pod.Status undefined (type string has no field or method Status)
!!! Error in /root/workspace/src/github.com/GoogleCloudPlatform/kubernetes/hack/lib/golang.sh:383
'go install "${goflags[@]:+${goflags[@]}}" -ldflags "${version_ldflags}" "${nonstatics[@]:+${nonstatics[@]}}"' exited with status 2
Call stack:
1: /root/workspace/src/github.com/GoogleCloudPlatform/kubernetes/hack/lib/golang.sh:383 kube::golang::build_binaries_for_platform(...)
2: /root/workspace/src/github.com/GoogleCloudPlatform/kubernetes/hack/lib/golang.sh:521 kube::golang::build_binaries(...)
3: ./hack/build-go.sh:26 main(...)
Exiting with status 1
!!! Error in /root/workspace/src/github.com/GoogleCloudPlatform/kubernetes/hack/lib/golang.sh:439
'kube::golang::build_binaries "$@"' exited with status 1
Call stack:
1: /root/workspace/src/github.com/GoogleCloudPlatform/kubernetes/hack/lib/golang.sh:439 kube::golang::build_binaries(...)
2: ./hack/build-go.sh:26 main(...)
Exiting with status 1
```

jsafrane commented:
It needs a rebase due to PR #11741, I'm working on it.

I got bitten by iSCSI not working through NAT created by the service - it exchanges real IP address of the server pod as part of the protocol.

jsafrane committed:
Add e2e iSCSI volume test.

The test container needs to run with --privileged, therefore it's disabled by
default.

jsafrane committed:
Add Ceph RBD e2e test.

The test assumes that all nodes have Ceph client utilities installed.
Ceph RBD container is hand crafted to be really minimal. It creates a new RBD
on startup, which can take up to several minutes on busy machines.

alex-mohr commented:
@jsafrane Please let us know when you get a chance to rebase!

brendandburns commented:
PR changed after LGTM, removing LGTM.

brendandburns commented:
PR changed after LGTM, removing LGTM.

mikedanese commented:
@brendanburns I think you broke github api. The PR didn't have LGTM label and you still managed to remove it twice.

jsafrane committed:
Revert "Use a service between e2e volume tests clients and servers."

iSCSI and RBD volumes don't work as Kubernetes services - these protocols
are broken by S-NAT created by kube-proxy - at least iSCSI exhanges real
IP address of the iSCSI target as part of the protocol.

This reverts commit 118004c16656617295234bc250da11788dec62c2.

jsafrane commented:
Sorry for the long wait, I had to revert commit 118004c and I wanted to make sure it really works in Vagrant and GCE. A patch that makes this test working on GCE will follow shortly (as a separate PR).

brendandburns commented:
LGTM.  Shippable is a flake, I kicked it again.

jsafrane commented:
Thanks for the merge!

@saad-ali, can you please build and push images from `test/images/volumes-tester/rbd` and `test/images/volumes-tester/iscsi` to gcr.io/google_containers?

saad-ali commented:
@jsafrane Will do

saad-ali commented:
@jsafrane Both `test/images/volumes-tester/rbd` and `test/images/volumes-tester/iscsi` images are pushed to `gcr.io google_containers`

```
...
5c8faa6d16fa: Image successfully pushed 
Pushing tag for rev [5c8faa6d16fa] on {https://gcr.io/v1/repositories/google_containers/volume-rbd/tags/latest}
Pushing tag for rev [5c8faa6d16fa] on {https://gcr.io/v1/repositories/google_containers/volume-rbd/tags/0.1}
...
93e4a66f9fc5: Image successfully pushed 
Pushing tag for rev [93e4a66f9fc5] on {https://gcr.io/v1/repositories/google_containers/volume-iscsi/tags/latest}
Pushing tag for rev [93e4a66f9fc5] on {https://gcr.io/v1/repositories/google_containers/volume-iscsi/tags/0.1}

```
